### DISCUSSION TITLE
Fedora compilation instructions are incorrect

### DISCUSSION THREAD

**Discussion by FSilveiraa** [YellowRoseCx/koboldcpp-rocm#74] [General]
I tried the setup on a completely new system with Fedora 40, following the instructions, and I noticed some of them were wrong:

- Besides `rocblas-devel` and `hipblas-devel` you also need the `rocm-hip-devel` package, otherwise you'll get an error about 2 missing header files: `hip/hip_fp16.h` and `hip/hip_runtime.h`
- I did not have to link anything on `/opt`, all libraries were found with no issue, but it was looking for the `amdgpu-arch` which in Fedora is `amdgpu-arch-17`. I fixed this by symlinking `/usr/bin/amdgpu-arch-17` to `~/.local/bin/amdgpu-arch`

That last issue only happens if you don't specify the GPU architecture, which you have to like the guide specifies, otherwise you'll get a runtime error after loading the model into memory. I can open a PR for the README, but I'd like someone else to confirm this if possible.

**Comment by rdnetto** [comment]
Ran into this when trying to debug building under Fedora 41.
Can confirm the above is correct, and creating `/opt/rocm` actually causes issues in finding `hipconfig` and should therefore be avoided.
